Marlin/src/feature/bedlevel/bedlevel.cpp

```diff
@@ -77,6 +81,11 @@ void set_bed_leveling_enabled(const bool enable/*=true*/) {
 
   if (can_change && enable != planner.leveling_active) {
 
+    // If a probe is present and the printer homes away from the bed, account for the probe offset.
+    #if HAS_BED_PROBE && Z_HOME_DIR > 0
+      current_position[Z_AXIS] += (enable ? -1 : 1) * zprobe_zoffset;;
+    #endif
```

Comment:
How does this affect `AUTO_BED_LEVELING_PLANAR` and `AUTO_BED_LEVELING_3POINT`?

Reply:
I'm starting to think that this should only apply to UBL since its meshes can persist across homing events. This may be true for some of the other leveling systems, but it seems like it wasn't when I was looking into them a few months ago.